src/reducers/index.js

```diff
@@ -10,11 +10,22 @@ const initialState = {
 // by our action creator. What does the reducer
 // need to do with the count in each case?
 export default (state = initialState, action) => {
+  console.log('reducer running', action)
   switch (action.type) {
     case INCREMENT:
+      return {count: state.count +1};
+
     // Fill in the body of this case
     case DECREMENT:
+      return Object.assign({}, state, {count: state.count-1});
     // Fill in the body of this case
+    
+    case INCREMENTIFODD:
+      return Object.assign({}, state, {count: state.count+2});
+    
+    case INCREMENTASYNC:
+      return {count: state.count +1};
```

Comment:
The async functionality would have to be implemented either in your action creator or in your Counter.js file